driver-core/src/main/com/mongodb/client/model/TextSearchOptions.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model;
+
+/**
+ * Text search options for the {@link Filters#text(String, TextSearchOptions)} helper
+ *
+ * @mongodb.driver.manual reference/operator/query/text $text
+ * @since 3.2
+ */
+public class TextSearchOptions {
+
+    private String language;
+    private Boolean caseSensitive;
```

Comment:
And the default would be false for that as well, and for the same reason (so as not to change behavior).  

I only care because Boolean is harder to use than boolean, so I avoid it if at all possible.  It sucks when:

    if (options.getCaseSensitive())

throws an NPE.  On the other hand, users will not call the getter normally.